app/controllers/users_controller.rb

```diff
@@ -40,10 +45,24 @@ def user_params
 
   def authorize
     authenticate_user!
-    if current_user.id.to_s == params[:id]
-      @user = current_user
-    else
-      redirect_to root_path
+    return redirect_to root_path, alert: I18n.t('denial.unauthorized') if current_user.nil?
+
+    if params.key?(:id)
```

Comment:
That's a lot of logic just for setting the `@user` for edit and update :sweat_smile:  
I would be fine with deleting this method, using `authenticate_user!` as before_action and maybe adding a method that just creates the instance variable if necessary.

```ruby
def user
  @user ||= User.find(params[:id])
end
```
It is already catched in the application controller if an ActiveRecord model cannot be found, so that we do not have to put this logic into every controller.

Reply:
> maybe adding a method that just creates the instance variable if necessary.
> 
> ```ruby
> def user
>   @user ||= User.find(params[:id])
> end
> ```

Nah, this smells like a possible cache invalidation problem to me if we are not careful :-)

> It is already catched in the application controller if an ActiveRecord model cannot be found, so that we do not have to put this logic into every controller.

Interesting point, on the other hand, the current solution is more convenient because you are redirected to the user list automatically ... What's more important for us, simple code or convenient error handling?

Reply:
IIRC an instance of a Controller is created for every request and deleted after that so caching shouldn't be a problem. Or did you had another case in mind?

> What's more important for us, simple code or convenient error handling?

That's a hard choice, I'm currently thinking if it is possible to parameterize the path but did not have an idea yet. So it's fine leaving it as it is :+1:

Reply:
> Or did you had another case in mind?

Not concretely, was only thinking of the rather complex solution in #86